datafusion/expr/src/expr.rs

```diff
@@ -1389,6 +1390,201 @@ impl Expr {
             | Expr::Placeholder(..) => false,
         }
     }
+
+    pub fn hash_node(&self, hasher: &mut AHasher) {
```

Comment:
what's the benefit of this vs using the derived `Hash` impl. I think a comment explaining the differences might be useful.

Reply:
Yeah this is a good point, I will add some comments here why CSE uses special hashing.

Reply:
Thanks, added.